### ISSUE TITLE
add JSONView response renderer

### ISSUE THREAD

**Issue by RandomFractals** [tanhakabir/rest-book#85]
for JSON response display with expand/collapse tree nodes for json documents, in addition to existing Data tab that should show formatted JSON doc.

see: https://github.com/tanhakabir/rest-book/issues/73#issuecomment-827007087

**Comment by RandomFractals** [comment]
@tanhakabir now that I know how to work that `</>` cell output renderer toggle, we can probably close this.

What I was looking for to see as a JSONView, which is one of the built-in mime type renderers :)

![image](https://user-images.githubusercontent.com/656833/116494360-44d6f200-a866-11eb-81c9-b52db8f3c537.png)

still would be nice to at least parse and format json data output in your custom Data tab:

![image](https://user-images.githubusercontent.com/656833/116494459-7b147180-a866-11eb-92b9-f59ba365f35d.png)

**Comment by tanhakabir** [comment]
yeah! I would still like to see a prettified view of the data in the Data tab for the custom output

**Comment by asmith3006** [comment]
Sorry I misunderstood a previous comment, I'll remove my previous post as it's confusing. 

I feel a built in renderer would be the best as it makes the response easier to read instantly without extra steps of copy/pasting. I use [RestClient](https://github.com/Huachao/vscode-restclient) a lot and the response is rendered appropriately. If we can combine the response rendering of RestClient with your notebooks I think we'd have a perfect extension.

**Comment by tanhakabir** [comment]
@asmith3006 just to clarify this version of the JSON view doesn't work well for you?

![recording (39)](https://user-images.githubusercontent.com/12758612/119170083-a1699f00-ba17-11eb-9ff9-003742ba9832.gif)

**Comment by RandomFractals** [comment]
@tanhakabir can you format json data response in your defualt Data tab without switching to built-in json renderer?

I think that's what we are missing to avoid extra clicks to get to that picker, and be able to quickly view data returned in json format.

Thanks!

**Comment by tanhakabir** [comment]
Good point! Sure!

**Comment by RandomFractals** [comment]
just an fyi I am using a restbook to test the leaflet map renderer I plan to work on. So far looks like we can use both just fine ;)

![leaflet-output-renderer](https://user-images.githubusercontent.com/656833/119292795-b3fdf700-bc16-11eb-9369-fbeb8315fccb.png)

**Comment by RandomFractals** [comment]
btw, you might want to try this: https://github.com/RandomFractals/vscode-data-table/issues/23#issuecomment-874833269

**Comment by tanhakabir** [comment]
cool!

**Comment by tanhakabir** [comment]
once you're happy with it, I can add your renderer as an optional dependency to my extension to view csvs

**Comment by RandomFractals** [comment]
good idea. I need to patch my stdout replacement and will let you know when that update is ready to use.

Thanks!

**Comment by RandomFractals** [comment]
@tanhakabir actually, you can add my data table renderer as a dependency for json and csv data display already.

I checked and my v1.1.0 should be able to parse and render both just fine. 

see: https://github.com/RandomFractals/vscode-data-table/releases/tag/v1.1.0

**Comment by RandomFractals** [comment]
... now with geojson data display too :wink: https://github.com/RandomFractals/vscode-data-table/releases/tag/v1.3.0

![world-rivers-geo-data-table](https://user-images.githubusercontent.com/656833/126359295-adf71cec-c54d-4090-8a36-15d2f32d0e1f.png)

**Comment by connor4312** [comment]
I've used https://www.npmjs.com/package/react-json-view in the past with success.

Imo a tabular view of json data is domain specific towards data science work and isn't nearly as common as people working with traditional REST APIs, so (unlike some degree of sane formatting of JSON responses 😉 ) I don't feel that such functionality needs to be built in.

**Comment by RandomFractals** [comment]
@connor4312 I think it would be best if vscode provided an api to integrate built-in json, xml and other text editor views in notebook cell renderers, so we can use them for uniform user experience and focus more on creating custom renderers like that data table, which I think with some work can be used by other devs for REST response views, not just data scientists.

I've seen a number of data tables rolled into different notebook platforms in vscode, R & Py, and some attempt to standardize that which varying results.
